### PR TITLE
add promise support #19

### PR DESCRIPTION
**Pull request by mreinstein**

### CODE REVIEW

**Comment by mreinstein** [comment]
@mcwhittemore I avoided more drastic cleanup because I wasn't sure what minimum version of node you want to support.

**Comment by mreinstein** [comment]
I also struggled to get the mocha tests written in a way that made sense. Here's the test that I wrote to verify my changes didn't break things.

test.js:
```
var sgf = require('./index.js');


var filter = "ACDMRTUXB";
console.log(sgf());
console.log(sgf(filter));

var invocationCount = 0;
console.log(sgf(function () { console.log('a') }));
console.log(sgf(filter, function () { console.log('b') }));
```

Expected output:

```bash
Promise { <pending> }
Promise { <pending> }
undefined
undefined
a
b
```

**Comment by mcwhittemore** [comment]
@mreinstein - I've added two tests for promises. It looks like the filter case for promises isn't working correctly. Also, I really need to setup CI on this sometime.

File: test/eval.js [hunk 1]
```diff
@@ -201,6 +201,40 @@ describe("As a module", function() {
...
+
+                    moveFile({
+                        oldFileName: data.filename,
+                        newFileName: newFileName
+                    }, function(err) {
+                        var sgf = newSGF();
+                        sgf('A').then(function(results) {
+                            results.length.should.equal(1);
+                            results[0].filename.should.equal(newFileName);
+                            results[0].status.should.equal("Added");
```
Comment (mreinstein) [inline test/eval.js]: don't we need a `done()` call here?

**Comment by mreinstein** [comment]
@mcwhittemore I think there might be a bug in the test: 

https://github.com/mcwhittemore/staged-git-files/commit/8339635b04218c3772ae5b2f02fe8d6b929535ee#r33436520

I also added the travis ci configuration, so once you enable this for your repo it should just start working.

**Comment by mcwhittemore** [comment]
Great. Thanks. All is passing. Will merge and release.

image

**Comment by mreinstein** [comment]
@mcwhittemore if you log in to circle-ci and enable `staged-git-files` repo for testing, you'll see the build badge and everything setup already. Let me know if you have any questions on this

**Comment by mcwhittemore** [comment]
This has been released in 1.2.0

**Comment by mcwhittemore** [comment]
> if you log in to travis-ci and enable

Have done. Thanks for over coming my laziness :)

**Comment by mreinstein** [comment]
btw, you might want to add node 11 and 12 to the `.travis.yml`. I forgot 12 is the latest!

**Pull request merged**